site/content/protocols/survey/0.1/readme.md

```diff
@@ -0,0 +1,179 @@
+```json
+{
+    "@type": "https://didcomm.org/survey/0.1/request",
+    "@id": "8192855c-89f3-5bb5-4971-7be10cbc6c71",
+    "~thread": {
+        "thid": "5689db78-5123-2aad-448d-0203107fee11"
+    },
+    "request": {
+        "survey_id": "750d9731-562b-f8a9-48df-89b12a1ec7f3",
+        "survey_schema": "json string defining the survey schema with the questions, answers and validations",
+        "ui_schema": "json string defining the ui schema such as layout, controls, help text to aid rendering of the survey",
+        "init_data": "json string containing the data to prefill answers",
+        "i18n_data": "json string containing the translations"
+    },
+    "expires_time": "2018-12-13T17:29:06+0000"
+}
+```
+
+The items in the message are as follows:
+
+- `@type` -- required, must be as above
+- `@id` -- required, must be as defined in [RFC 0005 DIDComm]
+- `~thread` -- required, must be as defined in [RFC 0008 Message ID and Threading]
+- `request` -- required, an item containing a JSONForms JSON structures. must be a single JSONForms request
+    - `survey_id` -- required, contains the survey identifier to uniquely identify the survey in the business logic
+    - `survey_schema` -- required, contains the questions, answers and validations
+    - `ui_schema` -- required, contains layout, controls, help labels
+    - `init_data` -- optional, it can be used to prefill/preselect answers for questions in the survey
+    - `i18n_data` -- optional, contains translations for the questions, answer options and labels
+- `expires_time` is optional
+
+
+### Survey Response Message
+
+A `response` message is sent by the `responder` following the completion of the survey to convey the answers to the requester.
+
+If the `request` is unrecognizable as a JSONForms survey such that a survey cannot be rendered, the server SHOULD send a [RFC 0035 Report Problem] message to the client.
+
+DIDComm V1 Example:
+```json
+{
+    "@type": "https://didcomm.org/survey/0.1/request",
+    "@id": "8192855c-89f3-5bb5-4971-7be10cbc6c71",
+    "~thread": {
+        "thid": "5689db78-5123-2aad-448d-0203107fee11"
+    },
+    "response": {
+        "response_type": "data",
+        "data": "json string containing the answers given by the responder",
+    },
+}
+```
+
+The items in the message are as follows:
+
+- `@type` -- required, must be as above
+- `@id` -- required, must be as defined in [RFC 0005 DIDComm]
+- `~thread` -- required, must be as defined in [RFC 0008 Message ID and Threading], same as the request
+- `request` -- required, an item containing a JSONForms JSON structures. must be a single JSONForms request
+    - `response_type` -- required, the type of response, must be `data` if the survey was completed and returns the answer data. If the responder declines the survey the `response_type` must be `decline`
+    - `data` -- required, if `response_type` is `data` and will contain the answers provided by the responder, optional with other `response_type`'s
```

Comment:
Do you think the problem report could be adopted into this protocol for declining a survey? I think response_type decline is fine as well, just thinking whether it could have advantages (we could define multiple codes for decline reasons).

Reply:
It could, but I think the response_type offers a bit more flexibility to define custom response types. At this stage both could work, but I've chosen for flexibility.